src/components/FormBase.tsx

```diff
@@ -0,0 +1,80 @@
+import { InputAdornment, TextField, Typography } from '@mui/material'
+import 'react-toastify/dist/ReactToastify.css'
+import { useForm, SubmitHandler, Resolver } from 'react-hook-form'
+import { FaAngleRight } from 'react-icons/fa6'
+import { useState } from 'react'
+
+interface FieldsConf {
+	name: string
+	label: string
+}
+interface FormBaseProps {
+	title?: string
+	subtitle?: string
+	size: 'small' | 'medium'
+	className: string | undefined
+	fields: FieldsConf[]
+	resolver: Resolver<any>
+	onSubmitSuccess: (data: any) => void
+	children?: React.ReactNode
+}
+
+const FormBase = ({ title, subtitle, size, className, fields, resolver, onSubmitSuccess, children }: FormBaseProps) => {
+	const [focusedField, setFocusedField] = useState<string | null>(null)
+	const {
+		register,
+		handleSubmit,
+		formState: { errors },
+		clearErrors,
+	} = useForm<any>({
```

Comment:
Pls avoid using `any`. This way you disable type checking

Reply:
resolved in account-settings branch